crates/gpui/src/platform/blade/shaders.wgsl

```diff
@@ -197,14 +274,70 @@ fn blend_color(color: vec4<f32>, alpha_factor: f32) -> vec4<f32> {
     return vec4<f32>(color.rgb * multiplier, alpha);
 }
 
+fn gradient_color(background: Background, position: vec2<f32>, bounds: Bounds,
+    sold_color: vec4<f32>, color0: vec4<f32>, color1: vec4<f32>) -> vec4<f32> {
+    var background_color = vec4<f32>(0.0);
+
+    switch (background.tag) {
+        default: {
+            return sold_color;
+        }
+        case 1u: {
+            // Linear gradient background.
+            // -90 degrees to match the CSS gradient angle.
+            let radians = (background.angle % 360.0 - 90.0) * M_PI_F / 180.0;
+            var direction = vec2<f32>(cos(radians), sin(radians));
+            let stop0_percentage = background.colors[0].percentage;
+            let stop1_percentage = background.colors[1].percentage;
+
+            // Expand the short side to be the same as the long side
+            if (bounds.size.x > bounds.size.y) {
+                direction.y *= bounds.size.y / bounds.size.x;
+            } else {
+                direction.x *= bounds.size.x / bounds.size.y;
+            }
+
+            // Get the t value for the linear gradient with the color stop percentages.
+            let half_size = bounds.size / 2.0;
+            let center = bounds.origin + half_size;
+            let center_to_point = position - center;
+            var t = dot(center_to_point, direction) / length(direction);
+            // Check the direct to determine the use x or y
+            if (abs(direction.x) > abs(direction.y)) {
+                t = (t + half_size.x) / bounds.size.x;
+            } else {
+                t = (t + half_size.y) / bounds.size.y;
+            }
+
+            // Adjust t based on the stop percentages
+            t = (t - stop0_percentage) / (stop1_percentage - stop0_percentage);
+            t = clamp(t, 0.0, 1.0);
+
+            switch (background.color_space) {
+                default: {
+                    background_color = srgba_to_linear(mix(color0, color1, t));
```

Comment:
Since this is a gradient, colors are `non-linear sRGB` and converted to `linear sRGB` after mixing?

Reply:
I tested it several times here, it seems that the color value obtained by the previous algorithm should be sRGB, which needs to be converted to linear to get the correct color.

Reply:
Oooh, okay, yeah I found the "previous algorithm", now it makes sense.